lib/index.js

```diff
@@ -2,35 +2,40 @@
 
 const Boom = require('boom');
 const Hoek = require('hoek');
+const Joi = require('joi');
 
 // Declare Internals
 
 const internals = {};
 
-exports.register = (server, options, next) => {
-
-    server.auth.scheme('bearer-access-token', internals.implementation);
-    next();
+internals.defaults = {
+    accessTokenName: 'access_token',
+    allowQueryToken: true,
+    allowCookieToken: false,
+    allowMultipleHeaders: false,
+    allowChaining: false,
+    tokenType: 'Bearer'
 };
 
-exports.register.attributes = {
-    pkg: require('../package.json')
-};
+internals.schema = Joi.object().keys({
+    validateFunc: Joi.func().required(),
+    accessTokenName: Joi.string().required(),
+    allowQueryToken: Joi.boolean(),
+    allowCookieToken: Joi.boolean(),
+    allowMultipleHeaders: Joi.boolean(),
+    allowChaining: Joi.boolean(),
+    tokenType: Joi.string().required()
+});
 
 internals.implementation = (server, options) => {
 
     Hoek.assert(options, 'Missing bearer auth strategy options');
```

Comment:
If we try keep the same error message *and* only use `joi` to produce it, it's actually a little complicated but for no real value add. I prefer code to be as simple to understand as possible.

Possible options without complicating things are either change error message to be `joi`s output, or keep using `Hoek.assert` here. I see no reason to change the error message, so happy to leave as is.